vendor_data.py

```diff
@@ -56,101 +52,190 @@ def __init__(self, VendorName, TargLang, WordRate, VendorMail = "", CatTool = "X
+        if type(SourceLang) != str:
+            raise TypeError("SourceLang should be a string!")
+        else:
+            self.SourceLang = SourceLang
         if type(VendorName) !=str:
             raise TypeError("VendorName should be a string!")
+        else:
+            self.VendorName = VendorName
+            
         if type(TargLang) != str:
             raise TypeError("TargLang should be a string!")
-        if type(WordRate) != float:
-            raise TypeError("WordRate should be a float!")
+        else:
+            self.TargLang = TargLang
+            
         if WordRate:
+            if type(WordRate) != float:
+                raise TypeError("WordRate should be a float!")
             if WordRate > 0.15:
-                raise ValueError("This vendor is too expensive, consider another one.")
-        elif WordRate == 0.00:
-            raise ValueError("Word rate cannot be 0.00.")
+                raise ValueError("This vendor is too expensive, pick another one.")
+            if WordRate == 0.00:
+                raise ValueError("Word rate cannot be 0.00.")
         else:
             self.WordRate = WordRate
-        if type(CatTool) != str:
-            raise TypeError("CatTool should be a string!")
-        if type(VendorMail) != str:
-            raise TypeError ("VendorMail should be a string!")
-        if VendorMail:   
+            
+        if Preferred != None:
+            if Preferred == True:
+                self.Status = "Preferred"
+            elif Preferred == False:
+                self.Status = "Back-up"
+        else:
+            self.Status = "Potential"
+            
+        if VendorMail: 
+            if type(VendorMail) != str:
+                raise TypeError ("VendorMail should be a string!")
             if(re.search(regex_mail,VendorMail)):
                 self.VendorMail = VendorMail
             else:
                 raise ValueError("Please insert a valid email address.")
         else:
             self.VendorMail = VendorMail
+            
+        if type(CatTool) != str:
+            raise TypeError("CatTool should be a string!")
+        else:
+            self.CatTool = CatTool
         if CatTool in VendorData.CatTools:
             self.CatTool = CatTool
         else:
             raise ValueError("This CAT tool is not valid. Run 'VendorData.CatTools' to check options.")
+        
+    def set_wordrate(self, NewRate):
+        if type(NewRate) != float:
+            raise TypeError("WordRate should be a float!")
+        elif NewRate > 0.15:
+            raise ValueError("This vendor is too expensive, consider another one.")
+        elif NewRate == 0.00:
+                raise ValueError("Word rate cannot be 0.00.")
+        else:
+            self.WordRate = NewRate
+
+    def set_status(self, PrefVend):
+        if PrefVend == True:
+            self.Status = "Preferred"
+        elif PrefVend == False:
+            self.Status = "Back-up"
+        else:

```

Comment:
you don't check what kind of values you have? `set_status("Preferred")` will set the status to "Potential"...